modules/runner/src/integrationTest/java/org/apache/ignite/internal/configuration/storage/ItRebalanceDistributedTest.java

```diff
@@ -534,13 +555,121 @@ void testDestroyPartitionStoragesOnRestartEvictedNode(TestInfo testInfo) throws
         checkInvokeDestroyedPartitionStorages(newNode, TABLE_NAME, 0);
     }
 
+    @Test
+    void testRebalanceWithTheSameNodes() throws Exception {
+        Node node = getNode(0);
+
+        createZone(node, ZONE_NAME, 1, 1);
+
+        clearSpyInvocations();
+
+        createTable(node, ZONE_NAME, TABLE_NAME);
+
+        assertTrue(waitForCondition(() -> getPartitionClusterNodes(node, 0).size() == 1, AWAIT_TIMEOUT_MILLIS));
+
+        alterZone(node, ZONE_NAME, 3);
+
+        checkAssignments();
+
+        updateAssignments();
```

Comment:
I would prefer to more specific naming, like `directUpdateMetastoreRebalanceAssignmentKeys` to show the real intention of this update.

Reply:
Fixed.